shell/platform/android/io/flutter/plugin/editing/SpellCheckPlugin.java

```diff
@@ -0,0 +1,158 @@
+
+  private final SpellCheckChannel mSpellCheckChannel;
+  private final TextServicesManager mTextServicesManager;
+  private SpellCheckerSession mSpellCheckerSession;
+
+  @VisibleForTesting MethodChannel.Result pendingResult;
+  @VisibleForTesting String pendingResultText;
+
+  // The maximum number of suggestions that the Android spell check service is allowed to provide
+  // per word. Same number that is used by default for Android's TextViews.
+  private static final int MAX_SPELL_CHECK_SUGGESTIONS = 5;
+
+  public SpellCheckPlugin(
+      @NonNull TextServicesManager textServicesManager,
+      @NonNull SpellCheckChannel spellCheckChannel) {
+    mTextServicesManager = textServicesManager;
+    mSpellCheckChannel = spellCheckChannel;
+
+    mSpellCheckChannel.setSpellCheckMethodHandler(this);
+  }
+
+  /**
+   * Unregisters this {@code SpellCheckPlugin} as the {@code
+   * SpellCheckChannel.SpellCheckMethodHandler}, for the {@link
+   * io.flutter.embedding.engine.systemchannels.SpellCheckChannel}, and closes the most recently
+   * opened {@code SpellCheckerSession}.
+   *
+   * <p>Do not invoke any methods on a {@code SpellCheckPlugin} after invoking this method.
+   */
+  public void destroy() {
+    mSpellCheckChannel.setSpellCheckMethodHandler(null);
+
+    if (mSpellCheckerSession != null) {
+      mSpellCheckerSession.close();
+    }
+  }
+
+  /**
+   * Initiates call to native spell checker to spell check specified text if there is no result
+   * awaiting a response.
+   */
+  @Override
+  public void initiateSpellCheck(
+      @NonNull String locale, @NonNull String text, @NonNull MethodChannel.Result result) {
+    if (pendingResult != null) {
+      result.error("error", "Previous spell check request still pending.", null);
+      return;
+    }
+
+    pendingResult = result;
+    pendingResultText = text;
+
+    performSpellCheck(locale, text);
+  }
+
+  /** Calls on the Android spell check API to spell check specified text. */
+  public void performSpellCheck(@NonNull String locale, @NonNull String text) {
+    String[] localeCodes = locale.split("-");
+    Locale localeFromString = LocalizationPlugin.localeFromString(locale);
+
+    if (mSpellCheckerSession == null) {
+      mSpellCheckerSession =
+          mTextServicesManager.newSpellCheckerSession(
+              null,
+              localeFromString,
+              this,
+              /** referToSpellCheckerLanguageSettings= */
+              true);
+    }
+
+    TextInfo[] textInfos = new TextInfo[] {new TextInfo(text)};
+    mSpellCheckerSession.getSentenceSuggestions(textInfos, MAX_SPELL_CHECK_SUGGESTIONS);
+  }
+
+  /**
+   * Callback for Android spell check API that decomposes results and send results through the
+   * {@link SpellCheckChannel}.
+   *
+   * <p>Spell check results will be encoded as a string representing the span of that result, with
+   * the format "start_index.end_index.suggestion_1/nsuggestion_2/nsuggestion_3", where there may be
```

Comment:
In the code, it is \n, not /n :)

Reply:
Thanks for pointing this out! I need to make another engine PR to remove my usage of `JSONMethodCodec`, so I'll fix this there.